src/common/function/src/scalars/timestamp/greatest.rs

```diff
@@ -34,21 +34,36 @@ pub struct GreatestFunction;
 
 const NAME: &str = "greatest";
 
-pub fn convert_to_date(arg: &str) -> Option<i32> {
+fn convert_to_date(arg: &str) -> Option<i32> {
     match Date::from_str(arg) {
         Ok(ts) => Some(ts.val()),
         Err(_err) => None,
     }
 }
 
-fn to_primitive_array(column: &VectorRef) -> Date32Array {
+fn string_vector_to_date32_array(column: &VectorRef) -> Date32Array {
     let column = StringVector::try_from_arrow_array(column.to_arrow_array()).unwrap();
     let column = (0..column.len())
         .map(|idx| convert_to_date(&column.get(idx).to_string()))
         .collect::<Vec<_>>();
     Date32Array::from_iter(column)
 }
 
+fn date_vector_to_date32_array(column: &VectorRef) -> Date32Array {
+    let column = DateVector::try_from_arrow_array(column.to_arrow_array()).unwrap();
```

Comment:
I think you can use `as_any` and `downcast_ref` to do this conversion:

```rust
column.as_any().downcast_ref::<DateVector>().context(CastDate)?;
```

And it's better to return an `Error` if cast fails.

Reply:
> And it's better to return an Error if cast fails.

Done. 

>I think you can use as_any and downcast_ref to do this conversion:

I couldn't address the review because one of the columns might be `ConstantVector`.

Actually, I got a error like this:
```
Failed to downcast vector: expect DateVector, but got ConstantVector
```

Reply:
Got it. You have to resolve it just like 

https://github.com/GreptimeTeam/greptimedb/blob/develop/src/common/function/src/scalars/aggregate/polyval.rs#L96-L101

Reply:
I guess that I addressed your review.